### ISSUE TITLE
Unexpected items were obtained in a subclass of ItemsControl

### ISSUE THREAD

**Issue by GochenRyan** [microsoft/microsoft-ui-xaml#9646]
### Describe the bug

I wrote a `RootDockControl` subclass of `ItemsControl`. But when I pass `DocumentDock` (its `VisibleDockables`) to it through binding, what it gets is `Document` (DocumentDock's `VisibleDockables`).

### Steps to reproduce the bug

1. Clone https://github.com/GochenRyan/WinUISample
2. Checkout commit a2089d3cdddd586a5eddff456d304fc3672aca02
3. Set Layout as startup project
4. insert a break point at https://github.com/GochenRyan/WinUISample/blob/a2089d3cdddd586a5eddff456d304fc3672aca02/WinUISample/Layout/Controls/RootDockControl.cs#L23
5. Run
<img width="599" alt="Snipaste_2024-05-20_23-58-57" src="https://github.com/microsoft/microsoft-ui-xaml/assets/15121614/e98cedeb-3034-4a88-a8c8-02eab6bd9be6">

### Expected behavior

The item should be `DocumentDock`

### NuGet package version

WinUI 3 - Windows App SDK 1.5.3: 1.5.240428000

### Windows version

Windows 11 (22H2): Build 22621

**Comment by GochenRyan** [comment]
I tested it to be an `ownerType` issue. I get the result I want when the `ownerType` of `VisibleDockablesProperty` is `DocumentDock`.  Is there a good way to handle the `ownerType` of `VisibleDockablesProperty` in the inheritance chain?

**Comment by ranjeshj** [comment]
@GochenRyan I'm having trouble understanding the issue. can you explain a bit more/provide a repro project that is scoped to the problem you are facing?

**Comment by GochenRyan** [comment]
@ranjeshj

**Comment by GochenRyan** [comment]
@ranjeshj 
I want to use the ContentTemplateSelector of ContentControl and the ItemTemplateSelector of ItemsControl to generate the corresponding custom controls step by step according to the node tree of xaml here：
```
<local1:DockControl>
    <models:RootDock>
        <models:RootDock.VisibleDockables>
            <models:DocumentDock>
                <models:DocumentDock.VisibleDockables>
                    <models:Document x:Name="Document1" Title="Program.cs">
                        <TextBlock Text="{Binding Title}"/>
                    </models:Document>
                    <!--<models:Document x:Name="Document2" Title="App.axaml">
                        <TextBlock Text="{Binding Title}"/>
                    </models:Document>-->
                </models:DocumentDock.VisibleDockables>
            </models:DocumentDock>
        </models:RootDock.VisibleDockables>
    </models:RootDock>
</local1:DockControl>
```
When creating a RootDockControl, its Items are not just the next level nodes.

**Comment by GochenRyan** [comment]
@ranjeshj Actually I have a solution. I created a new branch fix-containerdp and fixed this problem in commit 9b43b77802fdb0e097ce09c8f9e7aeefe27b43f8. In short, by putting the container dependency property at the last level of the inheritance chain. But I don't know why the dependency property of the container class needs to be handled specially.

**Comment by GochenRyan** [comment]
@ranjeshj Actually I have a solution. I created a new branch fix-containerdp and fixed this problem in commit 9b43b77802fdb0e097ce09c8f9e7aeefe27b43f8. In short, by putting the container dependency property at the last level of the inheritance chain. But I don't know why the container dependency property needs to be handled specially.

**Comment by GochenRyan** [comment]
@ranjeshj Actually I have a solution. I created a new branch fix-containerdp and fixed this issue in commit 9b43b77802fdb0e097ce09c8f9e7aeefe27b43f8. In short, by putting the container dependency property at the last level of the inheritance chain. But I don't know why the container dependency property needs to be handled specially.

**Comment by GochenRyan** [comment]
@ranjeshj My fault. I misunderstood the usage of Collection-type dp. FYI: https://learn.microsoft.com/en-us/windows/uwp/xaml-platform/custom-dependency-properties#collection-type-dependency-properties.
